### PR TITLE
PI-831 added pagination support for GET requests

### PR DESCRIPTION
**Pull request by thellimist**

### CODE REVIEW

**Comment by thellimist** [comment]
@jwineman merged.

File: src/API/AbstractAPIClient.php [hunk 1]
```diff
@@ -39,42 +39,74 @@ public function callAPI(Request $request)
...
+
+                if (json_last_error() !== JSON_ERROR_NONE) {
+                    throw new BadResponseException('Error decoding client API JSON', $response);
+                }
+
+                if (!$this->responseOk($response)) {
+                    $this->logAPICall($this->getAPIClientName(), array('type' => 'response', 'body' => $response), true);
+                }
+
+                if ($isPaginatable && isset($response['result_info'])) {
```
Comment (jwineman) [inline src/API/AbstractAPIClient.php:89]: These booleans are redundant - you can just check `isset($response['result_info'])`. Its impossible to have a pageable request in the V4 API without including this key.

File: src/API/AbstractAPIClient.php [hunk 2]
```diff
@@ -39,42 +39,74 @@ public function callAPI(Request $request)
...
+            $currentPage = 1;
+            $totalPages = 1;
 
-            // Since Guzzle automatically overwrites a new header when the request
-            // is POST / PATCH / DELETE / PUT, we need to overwrite the Content-Type header
-            // with setBody() function.
-            if ($method !== 'GET') {
-                $apiRequest->setBody(json_encode($request->getBody()), 'application/json');
-            }
+            while ($totalPages >= $currentPage) {
```
Comment (jwineman) [inline src/API/AbstractAPIClient.php:58]: Move this while loop into a function that takes a request with `result_info` and returns the merged result.

**Comment by jwineman** [comment]
Also I know this is my fault but lets move

```
Client($this->getEndpoint());
```

into the constructor and add a setter so we can test this class too.

**Pull request merged**